### PR TITLE
xilinx: gpio: add get_value implementation

### PR DESCRIPTION
**Pull request by amiclaus**
Add implementation for the `gpio_get_value` function in the Xilinx GPIO
driver.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: drivers/platform/xilinx/gpio.c [hunk 1]
```diff
@@ -382,12 +382,35 @@ int32_t gpio_set_value(struct gpio_desc *desc,
...
+			channel = 2;
+			pin -= 32;
+		} else
+			channel = 1;
+		*value = (XGpio_DiscreteRead(extra->instance, channel) >> pin) & 0x01;
+#endif
+		break;
+	case GPIO_PS:
+#ifdef XGPIOPS_H
+		*value = XGpioPs_ReadPin(extra->instance, pin);
```
Comment (mchindri) [inline drivers/platform/xilinx/gpio.c]: I see that you use pin here but is declared only when XGPIO_H is defined.
I think you can declare it outside the define.

File: drivers/platform/xilinx/gpio.c [hunk 2]
```diff
@@ -382,12 +382,35 @@ int32_t gpio_set_value(struct gpio_desc *desc,
...
-		// Unused variable - fix compiler warning
-	}
+	if (!desc)
+		return FAILURE;
 
-	if (value) {
-		// Unused variable - fix compiler warning
+	struct xil_gpio_desc	*extra;
+#ifdef XGPIO_H
+	uint8_t pin = desc->number;
```
Comment (mchindri) [inline drivers/platform/xilinx/gpio.c]: Nitpick: I think is better if we  declare the variables at the begging of the function

File: drivers/platform/xilinx/gpio.c (hunk 1 above)
Comment (amiclaus) [reply drivers/platform/xilinx/gpio.c]: You are correct. This is modified now.

**Comment by amiclaus** [comment]
V2:
- move `pin` variable declaration at the start of the function

**Pull request merged**